packages/expo/src/hooks/useSSO.ts

```diff
@@ -0,0 +1,104 @@
+import { useSignIn, useSignUp } from '@clerk/clerk-react';
+import type { EnterpriseSSOStrategy, OAuthStrategy, SetActive, SignInResource, SignUpResource } from '@clerk/types';
+import * as AuthSession from 'expo-auth-session';
+import * as Linking from 'expo-linking';
+import * as WebBrowser from 'expo-web-browser';
+
+import { errorThrower } from '../utils/errors';
+
+export type UseSSOParams = {
+  strategy: OAuthStrategy | EnterpriseSSOStrategy;
+  unsafeMetadata?: SignUpUnsafeMetadata;
+  redirectUrl?: string;
+};
+
+export type StartSSOParams = {
+  identifier?: string;
+  unsafeMetadata?: SignUpUnsafeMetadata;
+  redirectUrl?: string;
+};
+
+export type StartSSOFlowReturnType = {
+  /**
+   * Session ID created upon sign-in completion, or null if incomplete.
+   * If incomplete, use signIn or signUp for next steps like MFA.
+   */
+  createdSessionId: string | null;
+  setActive?: SetActive;
+  signIn?: SignInResource;
+  signUp?: SignUpResource;
+  authSessionResult?: WebBrowser.WebBrowserAuthSessionResult;
+};
+
+export function useSSO(useSSOParams: UseSSOParams) {
+  const { signIn, setActive, isLoaded: isSignInLoaded } = useSignIn();
+  const { signUp, isLoaded: isSignUpLoaded } = useSignUp();
+
+  async function startFlow(startSSOFlowParams: StartSSOParams = {}): Promise<StartSSOFlowReturnType> {
+    if (!isSignInLoaded || !isSignUpLoaded) {
+      return {
+        createdSessionId: null,
+        signIn,
+        signUp,
+        setActive,
+      };
+    }
+
+    let createdSessionId = signIn.createdSessionId;
+
+    const redirectUrl =
+      startSSOFlowParams?.redirectUrl ||
+      useSSOParams.redirectUrl ||
+      AuthSession.makeRedirectUri({
+        path: 'sso-native-callback',
+      });
+
+    await signIn.create({
+      strategy: useSSOParams.strategy,
+      redirectUrl,
+      identifier: startSSOFlowParams.identifier,
+    });
+
+    const { externalVerificationRedirectURL } = signIn.firstFactorVerification;
+    if (!externalVerificationRedirectURL) {
+      return errorThrower.throw('Missing external verification redirect URL for SSO flow');
+    }
+
+    const authSessionResult = await WebBrowser.openAuthSessionAsync(externalVerificationRedirectURL.toString());
+    if (authSessionResult.type !== 'success' || !authSessionResult.url) {
+      return {
+        authSessionResult,
+        createdSessionId,
+        setActive,
+        signIn,
+        signUp,
+      };
+    }
+
+    const parsedUrl = new URL(authSessionResult.url);
+    const rotatingTokenNonce = parsedUrl.searchParams.get('rotating_token_nonce') ?? '';
+    await signIn.reload({ rotatingTokenNonce });
+
+    if (signIn.firstFactorVerification.status === 'transferable') {
+      await signUp.create({
+        transfer: true,
+        unsafeMetadata: startSSOFlowParams?.unsafeMetadata || useSSOParams.unsafeMetadata,
+      });
+      createdSessionId = signUp.createdSessionId;
+    }
+
+    Linking.openURL(authSessionResult.url);
```

Comment:
This manually triggers the deep linking on the SSO callback since `WebBrowser.openAuthSessionAsync` is not triggering it. 

I'll fill a GitHub issue on Expo with a minimum reproduction for the above.